### PR TITLE
Use preferred PyPI name

### PR DESCRIPTION
**Pull request by jakirkham**
Closes https://github.com/conda-forge/msgpack-python-feedstock/issues/5

Upstream is pushing a rename of the PyPI package from `msgpack-python` to `msgpack`. So this makes that switch to the `sdist` source used. However the `conda` package is still called `msgpack-python`.

cc @hhromic @mingwandroid

xref: https://github.com/msgpack/msgpack-python/issues/280
xref: https://github.com/AnacondaRecipes/msgpack-python-feedstock/issues/1

### CODE REVIEW

**Comment by jakirkham** [comment]
Friendly nudge, @pmlandwehr. 😉

**Comment by jakirkham** [comment]
Will merge this by week's end if no comments.

**Comment by hhromic** [comment]
looks good to me! :)

**Comment by lesteve** [comment]
Just curious, what is the way for a change like this to reach the conda main channel? Is it to open a PR at https://github.com/ContinuumIO/anaconda-recipes?

**Comment by jakirkham** [comment]
So [AnacondaRecipes]( https://github.com/AnacondaRecipes/ ), which is what Anaconda (`defaults`) is now based off of since Anaconda 5.0, reuses conda-forge recipes. That doesn't mean they will take our changes blindly, but they will take a look at this change once accepted (if they haven't already) and evaluate integrating it.

File: recipe/meta.yaml [hunk 1]
```diff
@@ -1,21 +1,21 @@
...
 source:
   fn: {{ name }}-{{ version }}.{{ bundle }}
-  url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/{{ name }}-{{ version }}.{{ bundle }}
+  url: https://pypi.io/packages/source/{{ pypi_name[0] }}/{{ pypi_name }}/{{ pypi_name }}-{{ version }}.{{ bundle }}
   {{ hash_type }}: {{ hash }}
 
 build:
-  number: 0
+  number: 1
   script: python setup.py install --single-version-externally-managed --record=record.txt
```
Comment (carlodri) [inline recipe/meta.yaml]: should we switch to the recommended `pip install` method?

Comment (jakirkham) [reply recipe/meta.yaml]: Was trying not to go overboard on changes here (thought this one might be controversial). That said, you are right.

It should use `pip` and there are a few changes it needs for `conda-build` 3 support. I think I'll save that for a subsequent PR unless these are of particular importance to get in at the same time. Thoughts? :)

Comment (carlodri) [reply recipe/meta.yaml]: oh, you're right, now I see that dreaded `toolchain`... don't touch it..! 😉

Comment (jakirkham) [reply recipe/meta.yaml]: Should be addressed with PR ( https://github.com/conda-forge/msgpack-python-feedstock/pull/9 ).

**Pull request merged**